Multi category settings

feerrenrut opened the pull request:
There were a few issues with scaling / positioning of the settings dialog. I could not find a way to address this using WX (though https://github.com/wxWidgets/wxWidgets/pull/334 looks promising) instead, I have used windows API's to get the scale factor to scale hardcoded sizes.
I use `GetDpiForWindow` which is only available on Windows 10 build 1607 and later, and fallback to `GetDeviceCaps` which is available on Windows 7 and later.

Notice that the default size has changed, this is because I initially set this size while my scaling was set to 125%.

## Testing:
I have tried:
* Windows 10 build 1709 on a 1920x1080 monitor at 100%, 125%, 175%
* Windows 7 with SP1 VM with various monitor sizes and scalings.

## Known Issues:
The maximum size of the window is not limited, with a scaling factor high enough, the dialog can be larger than the available screen space.  In reality, I don't expect this to be a problem, and if it is, it will be a problem for several other NVDA dialogs as well. When testing this scenario in a Windows 7 VM, the Ok/Cancel button group was visible, and the resizing handle in the bottom right corner. I was able to shrink the dialog.

feerrenrut commented:
I forgot to mention Quentin has tried a build of this, and it resolves the issues for him.

leonardder commented on source/windowUtils.py:
```diff
@@ -94,6 +94,32 @@ def physicalToLogicalPoint(window, x, y):
+		if ret != 1:
+			log.ERROR("Unable to release the device context.")
+
+	if winDpi != 0:
```
May be add a simple comment here when the result would be 0?

feerrenrut commented:
I have updated the comments, and fixed a mistake with the case of  `log.ERROR` oops. This should be ready for review.

leonardder commented on source/windowUtils.py:
```diff
@@ -108,12 +108,16 @@ def getWindowScalingFactor(window):
+	# There is little information about what GetDeviceCaps does in the case of a failure for LOGPIXELSX, however,
+	# a value of zero is certainly an error.
 	if winDpi != 0:
 		return winDpi / DEFAULT_DPI_LEVEL
```
The comments you added above suggest that it might be good to log this as a debugWarning, since the return is now somehow obfuscating an error.

leonardder replied:
Woops, I placed the comment above below the wrong line of code. I mean log a debugWarning for winDpi == 0

feerrenrut commented:
Ok, I added better logging in there. I also moved those lines around so that the happy path is at the end of the function.

I have also fixed an issue that Quentin noticed with the Synth / Braille text controls not being large enough.
